pallets/staking/src/lib.rs

```diff
@@ -1964,12 +1999,27 @@ decl_module! {
             let ledger = Self::ledger(&controller).ok_or(Error::<T>::NotController)?;
             let stash = &ledger.stash;
 
-            ensure!(ledger.active >= <MinimumBondThreshold<T>>::get(), Error::<T>::InsufficientValue);
-            // Polymesh-Note - It is used to check whether the passed commission is <= commission cap.
-            ensure!(prefs.commission <= Self::validator_commission_cap(), Error::<T>::InvalidValidatorCommission);
+            // Polymesh-Note - Make sure stash has valid permissioned identity.
+            if let Some(id) = <Identity<T>>::get_identity(stash) {
+                // Ensure stash's identity is be permissioned.
+                let id_pref = Self::permissioned_identity(id)
+                    .ok_or_else(|| Error::<T>::StashIdentityNotPermissioned)?;
+                // Ensure identity doesn't run more validators than the intended count.
+                ensure!(id_pref.running_count < id_pref.intended_count, Error::<T>::HitIntendedValidatorCount);
+                ensure!(ledger.active >= <MinimumBondThreshold<T>>::get(), Error::<T>::InsufficientValue);
+                // Ensures that the passed commission is within the cap.
+                ensure!(prefs.commission <= Self::validator_commission_cap(), Error::<T>::InvalidValidatorCommission);
+
+                PermissionedIdentity::mutate(&id, |pref| {
+                    if let Some(p) = pref {
+                        p.running_count += 1
+                    }
+                });
```

Comment:
Since you already read the value a few lines above, using `insert` will be more optimal than `mutate` here.